pringles/simulator/simulator.py

```diff
@@ -41,26 +40,33 @@ def __init__(self, process_result, main_log_path=None, output_path=None):
         self.process_result = process_result
         self.main_log_path = main_log_path
         self.output_path = output_path
-        if self.successful():
+
+        if output_path:
             self.output_df = SimulationResult.parse_output_file(output_path)
+        if main_log_path:
             self.logs_dfs = SimulationResult.parse_main_log_file(main_log_path)
 
     def successful(self):
         return self.process_result.returncode == 0
 
-    @classmethod
-    def parse_output_file(cls, file_path):
-        return pd.read_csv(file_path, delimiter=r'\s+',
-                           names=[cls.TIME_COL, cls.PORT_COL, cls.VALUE_COL])
-
     @staticmethod
-    def _is_list_value(value: str) -> bool:
-        return value.strip().startswith("[") and value.strip().endswith("]")
+    def _parse_value(value: str):
+        is_list = value.strip().startswith("[") and value.strip().endswith("]")
+        if is_list:
+            return tuple(float(num) for num in value.replace('[', '').replace(']', '').split(', '))
+        return float(value)
 
-    @staticmethod
-    def _list_str_to_list(value: str) -> Tuple[float, ...]:
-        # We use tuple because it hay to be hashable to be used in a Dataframe
-        return tuple(float(num) for num in value.replace('[', '').replace(']', '').split(', '))
+    @classmethod
+    def parse_output_file(cls, file_path) -> pd.DataFrame:
```

Comment:
Should both parse methods be private?

Reply:
not sure, what if you have some old log files that you generated without pringles. Maybe you still want to parse them with the tool. For that case maybe we can decouple the results from the simulation (now the constructor receives the simulation proccess object and we almost don't use it).

Reply:
In that case, those method should be accessed that way. It's kind of weird for them to be exposed from SimulationResult.

For the time being, I would make them private.